restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverterService.java

```diff
@@ -0,0 +1,290 @@
+package io.stargate.web.docsapi.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.*;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.schema.Column;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+public class JsonConverterService {
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  public ObjectMapper getMapper() {
+    return mapper;
+  }
+
+  // TODO find a place to refactor this to, along with DocumentService#getBooleanFromRow
+  private static Boolean getBooleanFromRow(Row row, String colName, boolean numericBooleans) {
+    if (row.isNull("bool_value")) return null;
+    if (numericBooleans) {
+      byte value = row.getByte(colName);
+      return value != 0;
+    }
+    return row.getBoolean(colName);
+  }
+
+  public ImmutablePair<JsonNode, Map<String, List<JsonNode>>> convertToJsonDoc(
+      List<Row> rows, boolean writeAllPathsAsObjects, boolean numericBooleans) {
```

Comment:
Could we use simple `Pair` for the return type to make it more generic?

Reply:
Suggestion: use a custom inteface with `@Value.Immutable` for holding the result. It should be as efficient as the `ImmutablePair`, but more readable since we have assign meaningful names to the result components. It should able be easier to extend.